src/Waypoint/WaypointList.cpp

```diff
@@ -62,3 +62,30 @@ WaypointList::SortByDistance(const GeoPoint &location) noexcept
 {
   std::sort(begin(), end(), WaypointDistanceCompare(location));
 }
+
+class WaypointNameCompare
+{
+public:
+  explicit WaypointNameCompare() noexcept {}
+
+
+  [[gnu::pure]]
+  bool operator()(const WaypointListItem &a,
+                  const WaypointListItem &b) const noexcept {
+    return a.waypoint->name < b.waypoint->name;
+  }
+};
+
+void
+WaypointList::SortByName() noexcept
+{
+  std::sort(begin(), end(), WaypointNameCompare());
+}
+
+void
+WaypointList::MakeUnique() noexcept
+{
+  auto new_end = std::unique(begin(), end(), [](WaypointListItem &a, WaypointListItem &b)
+                                        {return a.waypoint->name == b.waypoint->name;});
```

Comment:
Why compare names? There could be two different waypoints with the same name (but different location, type etc.). That's not a good way to enforce unique waypoints, I think.

Reply:
I'm happy to use the waypoint id.  That, of course, would be unique.  I used the name because I felt that it would be reasonable to expect the name to be unique (NOT the shortname) and I also felt it would be most reasonable to present the list in some order and the most reasonable order was by name.

If I use waypoint id as the uniqueness, I would have to sort twice (once by id then unique() then by name).

Reply:
Rethinking it, if the name is unique, sorting by name would allow unique() to work as two with the same name would also have the same id and the second one would be eliminated even if we unique() on id!  As a bonus, we only sort once (by name).

Caution, though.  If, by chance, there were three with the same name in the list, two of which were the same id and the third a different waypoint, the list COULD contain a duplicate if the order was ID1, ID2, ID1.  I would consider this a very minor issue that would be acceptable as it is caused by non-unique names (which should be discouraged) and no data is lost to the user.

I have updated the PR to unique by id after sorting by name.